File: Rest-Api-Demo/controllers/greetingsController.js
let greetings = [
    { id: 1, recipient: "Jõuluvana", message: "Hüppa pommi!", sender: "Juss"},
    { id: 2, recipient: "Juss", message: "Ei hüppa!", sender: "Jõuluvana"},
    { id: 3, recipient: "Päkapikk", message: "Sellele Jussile küll enam kommi ei vii!", sender: "Jõuluvana"},
]

const getGreetingById = function name(id) {
    return greetings.find(x => x.id == id);
}

exports.getAll = (req, res) => {
    res.send(greetings)
}

exports.getById = (req, res) => {
    const greeting = getGreetingById(req.params.id)
    if (typeof greeting === 'undefined') {
        return res.status(404).send({ error: "Greeting not found" })
    }
    res.send(greeting)
}

exports.create = (req, res) => {
    if (!req.body.recipient || !req.body.message || !req.body.sender) {
        return res.status(400).send({ error: 'One or all params are missing' })
    }
    let newGreeting = {
        id: greetings[greetings.length - 1].id + 1,
        recipient: req.body.recipient,
        message: req.body.message,
        sender: req.body.sender,
    }
    greetings.push(newGreeting)
    res.status(201).location('localhost:8080/greetings/' + (greetings.length - 1)).send(
        newGreeting
    )
}

exports.update = (req, res) => {
    const greeting = getGreetingById(req.params.id)
    if (typeof greeting === 'undefined') {
        return res.status(404).send({ error: "Greeting not found" })
    }
    if (!req.body.recipient || !req.body.message || !req.body.sender) {
        return res.status(400).send({ error: "One or more params not found" })
    }
    greeting.recipient = req.body.recipient
    greeting.message = req.body.message
    greeting.sender = req.body.sender

    res.status(200).location('localhost:8080/greetings/' + greeting.id).send(greeting)
}

exports.delete = (req, res) => {
    const greeting = getGreetingById(req.params.id)
    if (typeof greeting === 'undefined') {
        return res.status(404).send({ error: "Greeting not found" })
    }
    greetings = greetings.filter(g => g.id !== greeting.id)
    res.status(204).send()
}

File: Rest-Api-Dotnet/Rest-Api-Dotnet/Controllers/GreetingsController.cs
using Microsoft.AspNetCore.Mvc;
using Rest_Api_Dotnet.Modules;

namespace Rest_Api_Dotnet.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class GreetingsController : ControllerBase
    {
        private static List<Greeting> _greetings = new List<Greeting>()
        {
            new Greeting() {Id = 1, Recipient = "Jõuluvana", Message = "Hüppa pommi!", Sender = "Juss"},
            new Greeting() {Id = 2, Recipient = "Juss", Message = "Ei hüppa!", Sender = "Jõuluvana"},
            new Greeting() {Id = 3, Recipient = "Päkapikk", Message = "Sellele Jussile küll enam kommi ei vii!", Sender = "Jõuluvana"},
        };

        // GET: api/<GreetingsController>
        [HttpGet]
        public IActionResult Get()
        {
            return new JsonResult(_greetings);
        }

        // GET api/<GreetingsController>/5
        [HttpGet("{id}")]
        public IActionResult Get(int id)
        {
            var result = _greetings.Find(x => x.Id == id);
            if (result == null)
            {
                return NotFound(new { Error = "Greeting not found" });
            }

            return new JsonResult(result);
        }

        // POST api/<GreetingsController>
        [HttpPost]
        public IActionResult Post([FromBody] Greeting newGreeting)
        {
            if (newGreeting == null)
            {
                return BadRequest();
            }

            var newId = _greetings.Last().Id + 1;
            newGreeting.Id = newId;
            _greetings.Add(newGreeting);

            return CreatedAtAction(nameof(Get), new { id = newId }, newGreeting);
        }

        // PUT api/<GreetingsController>/5
        [HttpPut("{id}")]
        public IActionResult Put(int id, [FromBody] Greeting greeting)
        {
            var result = _greetings.Find(x => x.Id == id);
            if (result == null)
            {
                return NotFound(new { Error = "Greeting not found" });
            }
            result.Recipient = greeting.Recipient;
            result.Message = greeting.Message;
            result.Sender = greeting.Sender;

            return AcceptedAtAction(nameof(Get), new { id = result.Id }, result);
        }

        // DELETE api/<GreetingsController>/5
        [HttpDelete("{id}")]
        public IActionResult Delete(int id)
        {
            var result = _greetings.Find(x => x.Id == id);
            if (result == null)
            {
                return NotFound(new { Error = "Greeting not found" });
            }
            _greetings.Remove(result);

            return NoContent();
        }
    }
}
